src/test/scala/mesosphere/marathon/state/GroupTest.scala

```diff
@@ -376,5 +376,26 @@ class GroupTest extends FunSpec with GivenWhenThen with Matchers {
       Then("the cycle is detected")
       current.hasNonCyclicDependencies should equal(false)
     }
+
+    it("can contain a path which has the same name multiple times in it") {
```

Comment:
Good call, such a test would be useless. I will do a new one using the current master as reference.